File: Homeworks/1. Programming/2. C#-Part-2/03.Methods/06.First larger than neighbours/FirstLargerThanNeighbours.cs
//Write a method that returns the index of the first element in array that is larger than its neighbours, or -1, if there is no such element
namespace _06.First_larger_than_neighbours
{
    using System;
    using System.Linq;

    class FirstLargerThanNeighbours
    {
        static int FindFirstLarger(int[] arr)
        {
            int counter = 0;

            for (int i = 1; i < arr.Length - 1; i++)
            {
                if (arr[i] > arr[i + 1] && arr[i] > arr[i - 1])
                {
                    counter = i; break;
                }
                else
                {
                    counter = -1;
                }
            }

            return counter;
        }

        static void Main()
        {
            int n = int.Parse(Console.ReadLine());
            var input = Console.ReadLine().Split(' ').Select(x => int.Parse(x)).ToArray();
            int index = FindFirstLarger(input);

            Console.WriteLine(index);
        }
    }
}

File: Homeworks/Web Design and UI technlogies/3. JavaScript-Part-1/06.Functions/06.First larger than neighbours/First larger than neighbours.js
/** Problem 6. First larger than neighbours
 * Write a method that returns the index of the first element in
 * array that is larger than its neighbours, or -1, if there is no such element.
 * Input                Output
 6
 -26 -25 -28 31 2 27    1
 */
function Larger(args) {
    var arrayLength = args[0],
        array = args[1].split(' ').map(Number),
        i;

    for (i = 0; i < arrayLength; i += 1) {
        if (array[i] < array[i + 1]) {
            return i;
        }
    }

    return -1;
}
